vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -2129,6 +2129,24 @@ def forward_native(
                 mode="constant",
                 value=0.0,
             )
+        do_naive_dispatch_combine: bool = (
+            self.dp_size > 1 and not self.quant_method.using_modular_kernel
+        )
+
+        def reduce_output(
+            states: torch.Tensor, do_combine: bool = True
+        ) -> torch.Tensor:
+            if do_naive_dispatch_combine and do_combine:
+                states = get_ep_group().combine(states, self.is_sequence_parallel)
+
+            if (
+                not self.is_sequence_parallel
+                and not self.use_dp_chunking
+                and self.reduce_results
+                and (self.tp_size > 1 or self.ep_size > 1)
+            ):
+                states = self.maybe_all_reduce_tensor_model_parallel(states)
+            return states
```

Comment:
Maybe we should move the naive dispatch call out to this level also.

Also, the original callsites for naive dispatch/combine are inside a sequence parallel context.  I'm not sure if that is going to cause problems.